### DISCUSSION TITLE
Is there a way to obtain the sql scripts with the changes applied to a client after calling the SynchronizeAsync method?

### DISCUSSION THREAD

**Discussion by marlonchosky** [Mimetis/Dotmim.Sync#1042] [Q&A]
Hi!
I have a case in which I have a server and a client databases with no communication between them (they are on different networks).
So a proposed solution was to put a copy of the client database in the same network of the server database, run the code to synchronize them and try to obtain the sql script that was applied in the moment the SyncAgent.SynchronizeAsync() method was called. In this way, I can save the script in a file and then carry it to the network where the client database lives and execute the script.

I'd like to know if there's a way to make my scenario true. Is it possible?

**Comment by Mimetis** [comment]
I think `DMS` is not meant to be used as a "Compare" databases tool.

In my opinion, the best solution for you would be to:
- Use DMS to sync your first client database (which have access to server)
- Make a comparison between the 2 clients databases using a tool like [dbComparer](https://dbcomparer.com/default.aspx)
- Apply the compare result changes to the second client database

**Reply by marlonchosky** [reply]
Hello @Mimetis, nice to talk to you :)
The thing is that I cannot use a tool like dbComparar because we synchronize the data between these two databases in a windows service, so the solution must be an automated solution.

 We already have a solution where we use DMS between 2 databases that are in the same network. 
 However, in this new scenario, those 2 databases can't communicate each other and IT won't do it. What their IT team suggest, is that we can produce, in some way, a file containing the changes made in the server to send to the client database (the changes goes only one way, from server database to client database).
 We came up with the idea initialize the two databases (client and server ) in the same network using DMS. Make a backup file of the client database (that already contains the initialization that DMS does) and move it to the other network (network where the client database must reside).

Then, we have 3 databases: Two databases in the network-1, where 1 database is the server and the other one is the client, and another database that resides in the network-2 and is only a copy from the database client that resides in network-1.

After this setup, we use DMS to synchronize the client database in network-1 using the server database that is in the same network-1, with the only difference that, besides doing the synchronization task that DMS does, we could produce an "object" (or collection of object) that represent the changes that DMS does when the next synchronization occurs. Then, we can serialize that object in some file, and the people will carry that file to the other network network-2.
Then, in the network-2 we can use DMS to load this serialized object to perform the changes in the client database.

**Is this scenario possible with DMS?**
Just for consideration, I've been reading the DMS docs and found [this](https://dotmimsync.readthedocs.io/Orchestrators.html#id2), where you can produce an object that contains the changes. 
Then I read [this](https://dotmimsync.readthedocs.io/Orchestrators.html#loadtablesfrombatchinfoasync) to try to load the tables from a batch, to try to load the changes but I got an exception. 

Am I going in the right direction? or simply this scenario is not possible to do with DMS?
